api4/user.go

```diff
@@ -1131,6 +1151,37 @@ func updateUser(c *Context, w http.ResponseWriter, r *http.Request) {
 	w.Write([]byte(ruser.ToJson()))
 }
 
+// checkProviderAttributes returns the empty string if the patch can be applied without
+// overriding attributes set by the user's login provider; otherwise, the name of the offending
+// field is returned.
+func checkProviderAttributes(a app.AppIface, ouser *model.User, patch *model.UserPatch) string {
```

Comment:
nit : Maybe we can just call it `user` or `originalUser` instead of `ouser` here?
```suggestion
func checkProviderAttributes(a app.AppIface, originalUser *model.User, patch *model.UserPatch) string {
```